jedi/plugins/stdlib.py

```diff
@@ -781,6 +781,77 @@ def _os_path_join(args_set, callback):
 }
 
 
+def infer_django_field(cls, field):
+    field_tree_instance, = field.infer()
+
+    if field_tree_instance.name.string_name in ('CharField', 'TextField', 'EmailField'):
+        model_instance_field_type, = cls.inference_state.builtins_module.py__getattribute__('str')
+        return DjangoModelField(model_instance_field_type, field).name
+
+    integer_field_classes = ('IntegerField', 'BigIntegerField', 'PositiveIntegerField', 'SmallIntegerField')
+    if field_tree_instance.name.string_name in integer_field_classes:
+        model_instance_field_type, = cls.inference_state.builtins_module.py__getattribute__('int')
+        return DjangoModelField(model_instance_field_type, field).name
+
+    if field_tree_instance.name.string_name == 'FloatField':
+        model_instance_field_type, = cls.inference_state.builtins_module.py__getattribute__('float')
+        return DjangoModelField(model_instance_field_type, field).name
+
+    if field_tree_instance.name.string_name == 'BinaryField':
+        model_instance_field_type, = cls.inference_state.builtins_module.py__getattribute__('bytes')
+        return DjangoModelField(model_instance_field_type, field).name
+
+    if field_tree_instance.name.string_name == 'BooleanField':
+        model_instance_field_type, = cls.inference_state.builtins_module.py__getattribute__('bool')
+        return DjangoModelField(model_instance_field_type, field).name
+
+    if field_tree_instance.name.string_name == 'DecimalField':
+        model_instance_field_type, = cls.inference_state.import_module(('decimal',)).py__getattribute__('Decimal')
+        return DjangoModelField(model_instance_field_type, field).name
```

Comment:
I think in general it would be nice to create something like

```
mapping = {'IntegerField': (None, 'str'), 'DecimalField': ('decimal', 'Decimal')}

def infer_django_field(...):
    try:
        module_name, attribute_name = mapping[field_tree_instance.name.string_name]
    except KeyError:
        pass
    else:
        if module_name is None:
            module = cls.inference_state.builtins_module
        else:
            module = cls.inference_state.import_module((module_name,))
        attribute, = module.py__getattribute__(attribute_name)
        return DjangoModelField(...).name
    # Handle foreign key stuff here.
```

Reply:
Don't you think if dictionary is overhead in this case? What about a NamedTuple or a class with defined `__slots__`.

Reply:
Speed doesn't matter at all here. Don't worry about it unless you profile and see that this is indeed an issue.

Or to say it in Knuth's words: "Premature optimization is the root of all evil" :)

Also you have to be careful in optimizing dicts away in Python. They are usually really fast and a dict access might be faster than all the comparisons that you are doing here, which sounds maybe counter intuitive, but dict access is really really fast, because it's what Python uses for everything. Comparisons (and math in general) are really slow, because of all the type checking. So intution can often be wrong for such things (also you are doing a ton of dict accesses anyway. `field_tree_instance.name.`, `cls.inference_state.builtins_module` and `.name is also one. So you have like 10 dict accesses anyway, which is definitely more than the one I proposed :).